.github/workflows/test.yaml

```diff
@@ -0,0 +1,53 @@
+name: Test
+on:
+  push:
+    branches:
+      - "**" # every branch
+      - "!gh-pages" # exclude gh-pages branch
+      - "!stage*" # exclude branches beginning with stage
+  pull_request:
+    branches:
+      - "**" # every branch
+      - "!gh-pages" # exclude gh-pages branch
+      - "!stage*" # exclude branches beginning with stage
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+    # enforce the same check as pre-commit
+    # but only run important checks
```

Comment:
Reuse pre-commit, but only run codespell, black and flake8 - doing exactly the same as previous [development.yaml](https://github.com/datajoint/datajoint-python/blob/3e3977147aedcef735561427664a4c7a81d3862f/.github/workflows/development.yaml#L92-L124)